### PR TITLE
osd: Use boost::intrusive::list, drop xlist

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

**Comment by mattbenjamin** [comment]
@shinobu-x I'm going to catch flack, but, I support   :) 👍

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
test this please

File: src/osd/PG.h [hunk 1]
```diff
@@ -810,7 +810,14 @@ class PG : public DoutPrefixProvider {
 
   /* You should not use these items without taking their respective queue locks
    * (if they have one) */
-  xlist<PG*>::item stat_queue_item;
```
Comment (cbodley) [inline src/osd/PG.h]: it doesn't look like `stat_queue_item` was used anywhere, i think we can just remove this one

File: src/osd/PrimaryLogPG.h [hunk 2]
```diff
@@ -692,7 +692,17 @@ class PrimaryLogPG : public PG, public PGBackend::Listener {
...
+  private:
+    boost::intrusive::list_member_hook<> queue_item_;
+  public:
+    typedef boost::intrusive::list<
+      RepGather,
+      boost::intrusive::member_hook<
+        RepGather,
+        boost::intrusive::list_member_hook<>,
+        &RepGather::queue_item_> > queue_item_t;
+    queue_item_t queue_item;
```
Comment (cbodley) [inline src/osd/PrimaryLogPG.h]: what you call `queue_item_t` here is the list type itself - can we call it `queue_t` instead?

with that in mind, it doesn't make sense for a queue item (`RepGather`) to itself contain a queue, so this unused member variable `queue_item_t queue_item;` should be removed

**Comment by cbodley** [comment]
@shinobu-x ping

**Comment by jdurgin** [comment]
Closing due to unaddressed comments

**Pull request closed without merging**